### ISSUE TITLE
UpdatePolicy on S3 Bucket

### ISSUE THREAD

**Issue by dscpinheiro** [aws-cloudformation/cfn-python-lint#1269]
cfn-lint version: 0.26.1 (installed with Homebrew)

*Description of issue*:
cfn-lint is asking for an [UpdatePolicy](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-attribute-updatepolicy.html) on a S3 Bucket (which is not applicable). It also looks like it should be ignored: https://github.com/aws-cloudformation/cfn-python-lint/blob/master/src/cfnlint/rules/resources/UpdateReplacePolicyDeletionPolicyOnStatefulResourceTypes.py#L40

Sample template:
```yaml
Resources:
  Logs:
    DeletionPolicy: Retain
    Type: AWS::S3::Bucket
    Properties:
      AccessControl: LogDeliveryWrite
      BucketEncryption:
        ServerSideEncryptionConfiguration:
          - ServerSideEncryptionByDefault:
              SSEAlgorithm: AES256
      PublicAccessBlockConfiguration:
        BlockPublicAcls: true
        BlockPublicPolicy: true
        IgnorePublicAcls: true
        RestrictPublicBuckets: true
```

Expected behavior: No output for `cfn-lint template.yaml`

Actual behavior:
```console
$ cfn-lint template.yaml 
W3011 Both UpdateReplacePolicy and DeletionPolicy are needed to protect resources from deletion
template.yaml:2:3
```

**Comment by PatMyron** [comment]
@dscpinheiro so that linked code snippet is actually from rule `I3011`:

https://github.com/aws-cloudformation/cfn-python-lint/blob/c3d16457059e03eaaf752f403d1983c1c7be3d0a/src/cfnlint/rules/resources/UpdateReplacePolicyDeletionPolicyOnStatefulResourceTypes.py#L11

whereas that message is from rule `W3011` getting flagged:

https://github.com/aws-cloudformation/cfn-python-lint/blob/c3d16457059e03eaaf752f403d1983c1c7be3d0a/src/cfnlint/rules/resources/BothUpdateReplacePolicyDeletionPolicyNeeded.py#L11

It's actually asking for an explicit [`UpdateReplacePolicy`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-attribute-updatereplacepolicy.html), not an [`UpdatePolicy`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-attribute-updatepolicy.html)

The idea behind `W3011` is that if you want to protect a resource from deletion with something like the `DeletionPolicy: Retain` above, it's likely you'd also want an [`UpdateReplacePolicy`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-attribute-updatereplacepolicy.html) to state that it should not be deleted during an update replacement either

**Comment by dscpinheiro** [comment]
Hi @PatMyron,

Sorry, I got confused and was looking at the wrong place.

I'll update my template to use _UpdateReplacePolicy: Retain_

Thanks!
